### DISCUSSION TITLE
How to keep track of cumulative number of simulated events?

### DISCUSSION THREAD

**Discussion by TDmetric** [metrumresearchgroup/mrgsolve#1108] [Q&A] [answered]
"Look at a stone cutter hammering away at his rock, perhaps a hundred times without as much as a crack showing in it. Yet at the hundred-and-first blow it will split in two, and I know it was not the last blow that did it, but all that had gone before."

I saw this quote a couple weeks ago and started thinking about how events like this could be modeled. I think what I'm trying to demo below could be called a joint Compound Poisson Process/Time to event (TTE) model? My issue is, I can't figure out how to keep track of the cumulative number of simulated events (strikes to the rock) to use as a covariate in the TTE model (stone breaking). Any thoughts from the community? Thank you in advance. TD

cptte <- '
$PLUGIN Rcpp

$PARAM @annotated
// counting process
TVWT  : 1/30    : typical wait time between events
TVLAM : 1       : lambda Poisson process

// time to event
TVLAM2 : 0.01  : lamba for exponential time-to-event model

$CMT  @annotated
HAZ     : hazard

$ODE
dxdt_HAZ = TVLAM2; // I want the cumulative sum of MAG to be a covariate on the hazard

$TABLE
// counting process
double ranexp = R::rexp(TVWT);
double cdf = R::pexp(ranexp, TVWT, 1, 0);
double ranun = R::runif(0, 1);
double event = ranun < cdf ? 1 : 0;
double MAG = R::rpois(TVLAM*event); // how can I keep track of the total events during the simulation?

// TTE
double CHAZ = HAZ;
double SUR = exp(-CHAZ);
double HAZNOW = TVLAM2;

$CAPTURE ranexp cdf ranun event MAG TVWT TVLAM TVLAM2
'

**Comment by kylebaron** [comment, marked as answer]
Hi Tyler - 

If you just want to keep track of cumulative MAG, you can try this:

- The code in `$MAIN` both initializes the variable and resets it to zero with each individual
- The code in `$TABLE`  accumulates `MAG`

Code isn't tested but let me know if this works for you. 

```r

$MAIN
if(NEWIND <= 1) double CUMAG = 0;

$TABLE
double MAG = R::rpois(TVLAM*event);
CUMAG = CUMAG + MAG;

```

**Reply by TDmetric** [reply]
So simple! It seems to work. Thanks so much. TD
